### ISSUE TITLE
Serialization of Infinity is null

### ISSUE THREAD

**Issue by mttrbrts** [accordproject/concerto#158]
**Describe the bug**
When an expression sets a value in the state to `Infinity`, this gets serialized (through `JSON.stringify`?) to `null`. This has the effect of invalidating the state instance and rendering the contract unable to accept new requests.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#Description
> The numbers Infinity and NaN, as well as the value null, are all considered null.

**To Reproduce**
infinity.ergo
```js
namespace matt.inf

import org.accordproject.cicero.contract.*
import org.accordproject.cicero.runtime.*

contract Inf over InfContract {
    clause infTest(request: Request): Response {
        set state InfContractState {
            stateId: "inf0",
            value: 1.0 / 0.0
        };
        return Response{}
    }
}
```

model.cto
```
namespace matt.inf

import org.accordproject.cicero.contract.* from https://models.accordproject.org/cicero/contract.cto
import org.accordproject.cicero.runtime.* from https://models.accordproject.org/cicero/runtime.cto

asset InfContractState extends AccordContractState {
    o Double value
}

asset InfContract extends AccordContract {}
```

state.json
```json
{
    "$class": "matt.inf.InfContractState",
    "stateId": "state1",
    "value": 0.0
}
```

contract.json
```json
{
    "$class": "matt.inf.InfContract",
    "contractId": "contract1"
}
```

request.json
```
{
    "$class": "org.accordproject.cicero.runtime.Request"
}
```

When you run ...
```bash
mattmbp:minimal matt$ ergorun execute --contract contract.json --request request.json --state state.json  model.cto infinity.ergo
09:05:31 - info:
{
  "clause": "mattXinfXInf",
  "request": {
    "$class": "org.accordproject.cicero.runtime.Request",
    "transactionId": "54280e38-bfda-43f1-98d1-9048a64d7aeb",
    "timestamp": "2019-08-16T08:05:31.493Z"
  },
  "response": {
    "$class": "org.accordproject.cicero.runtime.Response",
    "transactionId": "994be5fe-39b9-451e-9aa1-5e6a0fc90e3e",
    "timestamp": "2019-08-16T08:05:31.504Z"
  },
  "state": {
    "$class": "matt.inf.InfContractState",
    "value": null,
    "stateId": "inf0"
  },
  "emit": []
}
```
Notice the value of `value` in the state response.

If you then update the value of `state.json` with the new state, it will fail to pass validation.

```bash
mattmbp:minimal matt$ ergorun execute --contract contract.json --request request.json --state state.json  model.cto infinity.ergo
09:05:41 - error: Instance matt.inf.InfContractState#state1 missing required field value
```

**Expected behavior**
The engine should throw a runtime error, perhaps similar to an `enforce` response, to alert the client to the overflow and protect corruption of the state.

**Comment by jeromesimeon** [comment]
No activity on this for a while, I'm releasing the ownership for now.

**Comment by jeromesimeon** [comment]
Fix to this, changing the serializer `toJSON` to raise a validation error when the Double is `NaN`, `Infinity` or `-Infinity` is in PR #161

**Comment by jeromesimeon** [comment]
> A fix for this issue, by changing `toJSON` call in the serializer to raise a validation error when the Double is `NaN`, `Infinity` or `-Infinity` is available for review in PR #161.

That fix has been refactored to be done in the `resourcevalidator` rather than the `jsongenerator` class, as suggested by @mttrbrts. This provides better and more consistent error messages.
